syntax="proto3";
package message;

service VerificationService{
    rpc GetVerificationCode(GetVerificationRequest) returns (GetVerificationResponse){}
}

service ChattingServiceBalancer{
    //pass user's uuid parameter to the server, and returns available server address to user
    rpc AddNewUserToServer(RegisterToBalancer) returns (GetAllocatedChattingServer){}

    //user send SERVICE_LOGINSERVER request
    rpc UserLoginToServer(LoginChattingServer) returns (LoginChattingResponse){}
}

message GetVerificationRequest{
    string email = 1;
}

message GetVerificationResponse{
    int32 error = 1;
    string email = 2;
    string message = 3;
}

//register new uuid to balance and ask for chatting server allocation
message RegisterToBalancer{
    int32 uuid = 1;
}

//get server host & port & token of one chatting server
message GetAllocatedChattingServer{
    int32 error = 1;
    string host = 2;
    string port = 3;
    string token = 4;
}

//transfer uuid and token to balanceserver to check validation
message LoginChattingServer{
    int32 uuid = 1;
    string token = 2;
}

message LoginChattingResponse{
    int32 error = 1;
}